src/platform/datapath_epoll.c

```diff
@@ -76,7 +76,7 @@ typedef struct DATAPATH_RX_IO_BLOCK {
 
 } DATAPATH_RX_IO_BLOCK;
 
-typedef struct DATAPATH_RX_PACKET {
+typedef struct __attribute__((aligned(16))) DATAPATH_RX_PACKET {
```

Comment:
Can you add more info in your PR description as to why this helps and is necessary?

Reply:
Not 100% sure yet, so trying to run automation first

Reply:
Are you sure yet? 😄

Reply:
I did play with this little bit more and it is also sufficient (as far as I can tell) to increase `RecvBlockStride`

https://github.com/microsoft/msquic/blob/fbef931d274779bb02e635f8f4957a813126b911/src/platform/datapath_epoll.c#L333-L334

I originally added `+4` but it also works for me with +1 or +2. I sort of expected corrupted pointer if something overrides the location.

It seems like we are casting bunch of packets to single big buffer, right @nibanks? Are we positive the structures always land on aligned boundary? It seems like the issue is triggered only when we pack multiple packets into single buffer. In the test we look at with @ami-GS, there is single packet and then we have few together. As @ami-GS noticed it was always crashing on processing third one, e.g. first that is packed second in the buffer.

Reply:
@ami-GS can we shoot to get this merged this week? Please address my comment above, and associate this PR with the bug that's tracking the issue.